locking/locks.go

```diff
@@ -22,74 +22,88 @@ var (
 	ErrLockAmbiguous = errors.New("lfs: multiple locks found; ambiguous")
 )
 
-// LockFile attempts to lock a file on the given remote name
+// Client is the main interface object for the locking package
+type Client struct {
+	cfg       *config.Configuration
+	apiClient *api.Client
+	cache     *kv.Store
+}
+
+// NewClient creates a new locking client with the given configuration
+func NewClient(cfg *config.Configuration) (*Client, error) {
+
+	apiClient := api.NewClient(api.NewHttpLifecycle(cfg))
+
+	lockDir := filepath.Join(config.LocalGitStorageDir, "lfs")
+	os.MkdirAll(lockDir, 0755)
```

Comment:
Do you think we should let callers know about this error (even if it's non-fatal)?